tidy3d/plugins/design/design.py

```diff
@@ -45,39 +45,81 @@ class DesignSpace(Tidy3dBaseModel):
 
     name: str = pd.Field(None, title="Name", description="Optional name for the design space.")
 
+    task_name: str = pd.Field(
+        None,
+        title="Task Name",
+        description="Task name assigned to tasks along with a simulation counter. Only used when pre-post functions are supplied.",
+    )
+
+    path_dir: str = pd.Field(
+        ".",
+        title="Path Directory",
+        description="Directory for files to output to. Only used when pre-post functions are supplied.",
+    )
+
+    folder_name: str = pd.Field(
+        "default",
+        title="Folder Name",
+        description="Folder path where the simulation will be uploaded in the Tidy3D Workspace. Will use 'default' if no path is set.",
+    )
+
+    _dict_name: str = pd.PrivateAttr(default=None)
+
+    _task_names: list = pd.PrivateAttr(default=[])
```

Comment:
As a general philosophy, we avoid using mutable values within these tidy3d components. The reason for this is that it's generally much harder to reason about things if one has to keep track of state. 

I put some assorted thoughts about removing the generator and also the task_names list. it would probably save us a lot of headaches in the future based on my experience.